lib/spark/dsl/internal/insert_entities_into_sections.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Spark.Dsl.Internal.InsertEntitesIntoSections do
+  @moduledoc false
+  use Spark.Dsl.Transformer
+  alias Spark.Dsl.Transformer
+
+  def transform(dsl) do
+    e_and_s = dsl |> Transformer.get_entities([:top_level])
+
+    {entities, sections} =
+      Enum.reduce(e_and_s, {%{}, %{}}, fn e, acc ->
+        {entities, sections} = acc
+
+        case e do
+          %Spark.Dsl.Entity{name: name} ->
+            {Map.put(entities, name, e), sections}
+
+          %Spark.Dsl.Section{name: name} ->
+            {entities, Map.put(sections, name, e)}
+        end
+      end)
+
+    sections =
+      for {_, section} <- sections do
+        Map.put(section, :entities, Enum.map(section.entities, &Map.get(entities, &1)))
+      end
+
+    {:ok, dsl |> put_in([[:top_level], :entities], sections)}
```

Comment:
I opted to replace the function from before with a Transformer, since it seems to be more idiomatic Spark.

There wasn't a `Transformer` function for "replace all entities", so I opted to use `put_in/3` instead,
But this violates the warning in the `Transformer` moduledoc that says to only use `Transformer` functions to manipulate the DSL.

Reply:
Yeah, I'd suggest adding a function like `replace_entities` that takes a function of `list(entity) -> list(entity)`

Reply:
I'll create that function now and abstract the transformer to use it.